src/server/controlchan/commands/pasv.rs

```diff
@@ -39,10 +42,10 @@ impl Pasv {
     }
 
     #[tracing_attributes::instrument]
-    async fn try_port_range(local_addr: SocketAddr, passive_ports: Range<u16>) -> io::Result<TcpListener> {
+    fn try_port_range(local_addr: IpAddr, passive_ports: Range<u16>) -> io::Result<TcpSocket> {
```

Comment:
The only reason that it was `async` before is because it called `TcpListener::bind`, which also implicitly does `listen`.  But since I moved `listen` up out of this function, it no longer needs to be async.

Reply:
Thanks! Alright, then good. I think it's ready then. Could you rebase? Then we can have it merged.